tests/integration/test_select.py

```diff
@@ -115,3 +115,13 @@ def test_timezones(c, datetime_table):
     result_df = result_df.compute()
 
     assert_frame_equal(result_df, datetime_table)
+
+
+def test_date_casting(c, datetime_table):
```

Comment:
Thinks we don't actually need `datetime_table` here? Since we only use the context here

```suggestion
def test_date_casting(c):
```